src/datasets/features/features.py

```diff
@@ -1342,6 +1342,24 @@ def decode_nested_example(schema, obj, token_per_repo_id: Optional[Dict[str, Uni
     return obj
 
 
+_FEATURE_TYPES: Dict[Optional[str], FeatureType] = {}
+
+
+def register_feature(
+    feature_cls: type,
+    feature_type: Optional[str],
+):
+    """
+    Register a Feature object using a name and class.
```

Comment:
Maybe mention that this feature is experimental in the docstring ? In case we want to improve/change the API at one point

Reply:
should I use the experimental decorator also?

Reply:
Yes ideally :)